pkg/controller/dns/external_dns_crd.go

```diff
@@ -0,0 +1,109 @@
+package dns
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/controllername"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/metrics"
+	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/go-multierror"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ExternalDNSCRDControllerName = controllername.New("externaldns", "crd")
+
+type ExternalDNSCRDController struct {
+	config *config.Config
+	client client.Client
+	events record.EventRecorder
+}
+
+func NewExternalDNSCRDController(manager ctrl.Manager, config config.Config) error {
+	return ExternalDNSCRDControllerName.AddToController(ctrl.NewControllerManagedBy(manager).
+		For(&v1alpha1.ExternalDNS{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&corev1.ConfigMap{}), manager.GetLogger()).
```

Comment:
what's the point of owning the configmap? Just want to prevent user from changing it and bricking themselves?

Reply:
yep